ddm/src/db.rs

```diff
@@ -221,6 +332,31 @@ pub struct Ipv6Prefix {
     pub len: u8,
 }
 
+impl Ipv6Prefix {
+    pub fn db_key(&self) -> Vec<u8> {
+        let mut buf: Vec<u8> = self.addr.octets().into();
+        buf.push(self.len);
+        buf
+    }
+
+    pub fn from_db_key(v: &[u8]) -> Result<Self, Error> {
+        if v.len() < 5 {
+            Err(Error::DbKey(format!(
+                "buffer to short for prefix 4 key {} < 5",
+                v.len()
+            )))
+        } else {
+            Ok(Self {
+                addr: Ipv6Addr::from([
+                    v[0], v[1], v[2], v[3], v[4], v[5], v[6], v[7], v[8], v[9],
+                    v[10], v[11], v[12], v[13], v[14], v[15],
+                ]),
```

Comment:
```suggestion
                addr: Ipv6Addr::from(<[u8; 16]>::try_from(&v[..16]).unwrap()),
```
I couldn't quite get the cleaner `&v[..16].try_into()` to typecheck here, but this might be a bit less repetitive.

Reply:
Done